cluster/gce/manifests/kube-apiserver.manifest

```diff
@@ -11,6 +11,11 @@
 },
 "spec":{
 "securityContext": {
+    {{runAsUser}}
+    {{runAsGroup}}
+    {{capabilities}}
+    {{allowPrivilegeEscalation}}
```

Comment:
Is the `setcap`'d kube-apiserver able to successfully bind port 443 with this in place?

The documentation of no_new_privs explicitly calls out that one purpose is to prevent file capabilities from being effective. This matches the testing I just did in docker with --security-opt=no-new-privileges:true.

Reply:
Yup it seems to be working for me
here is the security context in the manifest
```
"securityContext": {
    "runAsUser": 2020,
    "runAsGroup": 2020,
    "capabilities": { "drop": ["all"], "add": ["NET_BIND_SERVICE"]},
    "allowPrivilegeEscalation": false,
    "supplementalGroups": [ 2042,2046 ],
    "seccompProfile": {
        "type": "RuntimeDefault"
    }
},
```
port flag setting
```
"--secure-port=443",
"--insecure-port=0",
```

Seems to be running
```
sudo crictl ps --name kube-apiserver
CONTAINER           IMAGE               CREATED             STATE               NAME                ATTEMPT             POD ID
0127d5bce806b       1e8af2c849269       8 minutes ago       Running             kube-apiserver      1                   1172ad27a5670
```

Looking at the process is self:
```
2020        4100 17.8 13.7 1303376 517856 ?      Sl   03:06   2:11 /usr/local/bin/kube-apiserver --allow-privileged=true --profiling=false --v=3 --cloud-co ...
```

Looking at Capabilities
```
sudo cat /proc/4100/status | grep Cap
CapInh: 00000000a80425fb
CapPrm: 0000000000000400
CapEff: 0000000000000400
CapBnd: 00000000a80425fb
CapAmb: 0000000000000000
```

Decoding the capabilities
```
capsh --decode=0000000000000400
0x0000000000000400=cap_net_bind_service
```

Reply:
That's... really interesting. I've reproduced your success. Thank you.